src/api/entities/Instruction/index.ts

```diff
@@ -550,6 +561,110 @@ export class Instruction extends Entity<UniqueIdentifiers, string> {
     };
   }
 
+  /**
+   * @hidden
+   *
+   * Retrieve all legs of this Instruction from chain
+   *
+   * @note supports pagination
+   *
+   * @throws if the instruction is executed/rejected and was pruned from chain (when querying from chain)
+   */
+  public async getLegsFromChain(paginationOpts?: PaginationOptions): Promise<ResultSet<Leg>> {
+    const {
```

Comment:
I also added this method to allow REST API to fetch legs and return those when an instruction is just created. Also, while testing middleware tests with different scenarios, it felt this method would be really useful (if someone doesn't have to wait for middleware to sync). I earlier refactored this to context class, but here the usage felt more suitable. 

FYI I have kept it as hidden (from docs) for now.